campaignion_foundation.js

```diff
@@ -156,4 +156,49 @@ Drupal.behaviors.payment_slide.attach = function (context, settings) {
   });
 };
 
+/**
+ * Provide an event to act on form step changes.
+ *
+ * If no form step is yet known, assume we are initially loading a page with a
+ * webform.
+ *
+ * You can use this event to conditionally show/hide parts of a page.
+ *
+ * NB: We currently need to use native events to allow dispatching between
+ * different versions of jQuery (`.trigger()` and `.on()` communication
+ * between the Drupal's `$` and the the mo-foundation-base's `$` does not
+ * work).
+ */
+Drupal.behaviors.form_step_data = {};
+Drupal.behaviors.form_step_data.attach = function (context, settings) {
+  // current step is always a number > 0 if the settings information is correct
+  var currentStep;
+  try {
+    currentStep = settings.campaignion_foundation.webform.current_step || 1;
+  }
+  catch {
+    // Return if we have no webform on the page
+    return
+  }
+
+  currentStep = parseInt(currentStep, 10);
+  var previousStep = $("#page").attr("data-form-step") || 0;
```

Comment:
Just realized: doesn’t this lead to "changeFormStep" always being triggered on the first page load? Is that expected behavior?

Reply:
Yes, expected. `if (previous < 1) {}` then we are on an initial page load. In the depending themes I check for that.

Reply:
Ok. But why not check for that here instead?

Reply:
I decided to include the indication of the initial state in this event as well. Otherwise I would need/want to trigger a different event on initialization. I did not see any specific benefit for this.

Reply:
I think neither "changeFormStep" nor "sameFormStep" should be triggered on initial page load because both don’t apply. Why would you need a custom event on initialization, you could use the native js load events?

Reply:
- Drupal behaviors trigger several times, but only 1 has current != previous in this logic
- it does not hurt to get the information: you can do all handler logic related to any steps changes in 1 handler and need not use a native JS load event, it's in 1 place. You get all the information currently needed in a single event.

Reply:
Please post a suggestion how to change the behavior. I still cannot see a compelling reason why to change the current state.

Reply:
Not sure what you are asking me here? :thinking: 

1. I'd expect "changeFormStep" to fire when the form step changes. On initial page load, the form step does not change.
2. I fail to see *why* you’d need a form step related event on initial page load, but if you think it *is* needed, I'd prefer an additional event, e.g. "initFormStep".
3. I guess the most proper solution would be to only do anything when `context` is the webform → should only fire events when the webform changes and only once per change. (+ the init event as discussed in 2. when `context == '#document'`)

Reply:
1. I would argue it changes from nothing to something
2. agree
3. then we would still need to add code that ensures events get only fired once. that would complicate stuff. which element would be the best to e.g. call `$.once()` on?

Reply:
:heart_eyes: 

Thank you! Now getting to a different form step as well as staying on the same step (e.g. due to a server side validation error) will trigger a "changeFormStep" event but I think this is fine. "Change" can also mean the form has changed (it was replaced via AJAX) but not necessarily the step we’re on.

Reply:
ah, yes. we would at least have the current/previous numbers to check for that then... :grin: